server.go

```diff
@@ -64,6 +64,75 @@ func NewServer(opt ...Option) (*Server, error) {
 	}, nil
 }
 
+// Index of rightmost occurrence of b in s.
+func last(s string, b byte) int {
+	i := len(s)
+	for i--; i >= 0; i-- {
+		if s[i] == b {
+			break
+		}
+	}
+	return i
+}
+
+// validateAddr will not only validate the addr, but if it's an ipv6 literal without
+// proper brackets, it will add them.
+func validateAddr(addr string) (string, error) {
+	const op = "gldap.parseAddr"
+
+	lastColon := last(addr, ':')
+	if lastColon < 0 {
+		return "", fmt.Errorf("%s: missing port in addr %s : %w", op, addr, ErrInvalidParameter)
+	}
+	rawHost := addr[0:lastColon]
+	rawPort := addr[lastColon+1:]
+	if len(rawPort) == 0 {
+		return "", fmt.Errorf("%s: missing port in addr %s : %w", op, addr, ErrInvalidParameter)
+	}
+	if len(rawHost) == 0 {
+		return fmt.Sprintf(":%s", rawPort), nil
+	}
+	// ipv6 literal with proper brackets
+	if rawHost[0] == '[' {
+		// Expect the first ']' just before the last ':'.
+		end := strings.IndexByte(rawHost, ']')
+		if end < 0 {
+			return "", fmt.Errorf("%s: missing ']' in ipv6 address %s : %w", op, addr, ErrInvalidParameter)
+		}
+		trimedIp := strings.Trim(rawHost, "[]")
+		if net.ParseIP(trimedIp) == nil {
+			return "", fmt.Errorf("%s: invalid ipv6 address %s : %w", op, rawHost, ErrInvalidParameter)
+		}
+		// ipv6 literal has enclosing brackets, and it's a valid ipv6 address, so we're good
+		return fmt.Sprintf("%s:%s", rawHost, rawPort), nil
+	}
+
+	// see if we're dealing with a hostname
+	hostnames, _ := net.LookupHost(rawHost)
```

Comment:
This triggers a DNS lookup and may be slow or may have unexpected consequences. I suggest `netip.ParseAddr` and see if there is an error.

Reply:
Hum... I converted to resolver.LookupHost(...) so I can add a timeout which will alievate some of the issues.
the point of looking up the host it to ensure we have an IP after this lookup which the rest of the func relies on.

Reply:
Ah. I thought that the point was to discover whether or not it's an IP address -- not that you needed an IP address at the end of the function...since Dial will do that resolution for you.

Reply:
If it's ipv6, we sort of "need" a proper literal IP (if it's an IP) otherwise the Listen(...) will fail.  Besides trying to form a proper literal ipv6, we're validating that addr is indeed something that we can pass along to the Listen(...) call. 

At this point, we want to resolve the hostname if needed, so we can rely on the fact that we're mucking around with an IP for the rest of the func.